src/main/java/com/zoo/animals/Animal.java

```diff
@@ -1,4 +1,49 @@
 package com.zoo.animals;
 
 public class Animal {
-}
+
+    private int energyLevel = 100;
+    protected String name;
+    protected int age;
+    protected int weight;
+
+    public Animal(int energyLevel, String name, int age, int weight) {
+        this.energyLevel = energyLevel;
```

Comment:
Here, there is a discrepancy - we have a default value for `energyLevel`, but at the same time, we force users to specify `energyLevel` explicitly when creating an object. This is now the only existing constructor, so the user has no other choice.

Possible solution - as we have a requirement to have a default value for energy level - remove this parameter from the constructor and don't set it in the constructor